src/base/periodic_boundary_base.C

```diff
@@ -64,6 +71,45 @@ bool PeriodicBoundaryBase::is_my_variable(unsigned int var_num) const
 }
 
 
+
+bool PeriodicBoundaryBase::has_transformation_matrix() const
+{
+  return (_transformation_matrix.get() != nullptr);
```

Comment:
Is this three times as wordy as necessary?  unique_ptr has a conversion to bool, and even if it didn't then the raw pointer would have a conversion to bool;

Reply:
I tried "return _transformation_matrix;" initially, but that didn't compile. What did you have in mind?

Reply:
Ah!  unique_ptr::operator bool is declared *explicit*, and that auto-applies in "if" and a dozen other cases but "return" isn't one of them.  ```return _transformation_matrix.get();``` should work, though, right?  That's what I usually do out of force of habit.  ```bool(_transformation_matrix)``` ought to be kosher, too, and is probably the most readable idiom.

Reply:
```
return !!_transformation_matrix;
```
I'm joking, but would that really be so bad?

Reply:
I changed it to bool(_transformation_matrix)

Reply:
We'll go with bool().

@jwpeterson:
Bang bang?  I shot you down.